### PR TITLE
76 create an endpoint for showing logged in users allocation

### PR DESCRIPTION
**Pull request by ywkim312**
This will show the allocation information of the logged in user. Test it by accessing `http://localhost:8080/space/api/allocations`

### CODE REVIEW

**Review by longshuicy — APPROVED** [review]
Looks good. Can we get this merged and deployed soon?
I'd like to work on its corresponding frontend task

File: server/space-service/src/main/java/edu/illinois/ncsa/incore/service/space/controllers/AllocationsController.java [hunk 1]
```diff
@@ -66,12 +66,21 @@ public class AllocationsController {
...
+    public String getUsage(@HeaderParam("x-auth-userinfo") String userInfo) {
+        JSONObject outJson = null;
+
+        String username = JsonUtils.parseUserName(userInfo);
+
+        try {
+            outJson = JsonUtils.createUserFinalQuotaJson(username, finalQuotaRepository);
+        } catch (ParseException e) {
+            logger.error("Error extracting allocation");
+            throw new IncoreHTTPException(Response.Status.INTERNAL_SERVER_ERROR, "Error extracting allocation");
```
[the remark this answers is not in the document]
Comment (navarroc) [reply server/space-service/src/main/java/edu/illinois/ncsa/incore/service/space/controllers/AllocationsController.java:81]: If it's rare and shouldn't happen, then I think 500 is ok. 500 should be reserved for odd errors that shouldn't happen.

**Review by nunezco2 — APPROVED** [review]

**Comment by ywkim312** [comment]
> Is the default return really 20 GB? I tried passing a random username to my local and it returns:
> 
> ```json
> {
> "total_number_of_datasets": 1000,
> "total_number_of_dfr3": 1000,
> "total_number_of_hazards": 1000,
> "total_file_size_of_datasets_byte": 21474836480,
> "total_file_size_of_hazard_datasets_byte": 21474836480,
> "total_file_size_of_hazard_datasets": "20.0 GB",
> "total_number_of_hazard_datasets": 1000,
> "total_file_size_of_datasets": "20.0 GB",
> "user": "test-user"
> }
> ```
> 
> Does that look right?

This is the values that I got from you and Gowtham. It was like 10 times for the previous values. This is for up until we have the user group real time update being fired. After that, this value will probably be meaningless

**Comment by ywkim312** [comment]
> Is the default return really 20 GB? I tried passing a random username to my local and it returns:
> 
> ```json
> {
> "total_number_of_datasets": 1000,
> "total_number_of_dfr3": 1000,
> "total_number_of_hazards": 1000,
> "total_file_size_of_datasets_byte": 21474836480,
> "total_file_size_of_hazard_datasets_byte": 21474836480,
> "total_file_size_of_hazard_datasets": "20.0 GB",
> "total_number_of_hazard_datasets": 1000,
> "total_file_size_of_datasets": "20.0 GB",
> "user": "test-user"
> }
> ```
> 
> Does that look right?

If you provide me the new number I can change it. Maybe let's come back to previous value, which is 1/10 of this value?

**Pull request merged**